src/llm/continuous_batching/servable_initializer.cpp

```diff
@@ -0,0 +1,196 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <vector>
+
+#include <openvino/genai/continuous_batching_pipeline.hpp>
+#include <openvino/openvino.hpp>
+#include <spdlog/spdlog.h>
+
+#include "../../json_parser.hpp"
+#include "../../logging.hpp"
+#include "../../status.hpp"
+#include "servable.hpp"
+
+#pragma warning(push)
+#pragma warning(disable : 4005 4309 6001 6385 6386 6326 6011 4005)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#include "mediapipe/framework/calculator_graph.h"
+#pragma GCC diagnostic pop
+#pragma warning(pop)
+
+#include "../../mediapipe_internal/mediapipe_utils.hpp"
+#include "servable_initializer.hpp"
+#include "src/llm/text_processor.hpp"
+
+namespace ovms {
+
+ov::genai::SchedulerConfig ContinuousBatchingServableInitializer::prepareDraftPipelineSchedulerConfigExperimental(const mediapipe::LLMCalculatorOptions_PipelineConfig& draftPipelineConfig) {
+    ov::genai::SchedulerConfig config;
+    config.max_num_batched_tokens = draftPipelineConfig.max_num_batched_tokens();
+    config.cache_size = draftPipelineConfig.cache_size();
+    config.dynamic_split_fuse = draftPipelineConfig.dynamic_split_fuse();
+    config.max_num_seqs = draftPipelineConfig.max_num_seqs();
+    config.enable_prefix_caching = draftPipelineConfig.enable_prefix_caching();
+    return config;
+}
+
+ov::genai::SchedulerConfig ContinuousBatchingServableInitializer::prepareDraftPipelineSchedulerConfig(const mediapipe::LLMCalculatorOptions& nodeOptions) {
+    ov::genai::SchedulerConfig config;
+    config.max_num_batched_tokens = nodeOptions.has_draft_max_num_batched_tokens() ? nodeOptions.draft_max_num_batched_tokens() : nodeOptions.max_num_batched_tokens();
+    config.cache_size = nodeOptions.has_draft_cache_size() ? nodeOptions.draft_cache_size() : nodeOptions.cache_size();
+    config.dynamic_split_fuse = nodeOptions.has_draft_dynamic_split_fuse() ? nodeOptions.draft_dynamic_split_fuse() : nodeOptions.dynamic_split_fuse();
+    config.max_num_seqs = nodeOptions.has_draft_max_num_seqs() ? nodeOptions.draft_max_num_seqs() : nodeOptions.max_num_seqs();
+    config.enable_prefix_caching = nodeOptions.enable_prefix_caching();
+    return config;
+}
+
+Status ContinuousBatchingServableInitializer::initializeExperimental(std::shared_ptr<GenAiServable>& servable, const mediapipe::LLMCalculatorOptions& nodeOptions, std::string graphPath) {
+    auto continousBatchingPipelineConfig = nodeOptions.continuous_batching_pipeline_config();
+    auto mainPipelineConfig = continousBatchingPipelineConfig.main_pipeline_config();
```

Comment:
why do we need to have pairs: servable<->initializer? the initializer only exists because it needs to be used in servable - why cant it be part of servable? is it separate the code for clarity? if we had it inside servable class, we could be using the actual this->properties with known derived type

Reply:
Mostly for code separation (less code per file, easier to find stuff) as well as logical separation (initializer is OVMS core thing and has dependency on Mediapipe, while servables are used only in the calculator and potentially could be initialized in a different way (for example if we extract it from the MP framework for some reason).